packages/eslint-plugin/lib/rules/prefer-telephone-component.js

```diff
@@ -0,0 +1,56 @@
+const jsxAstUtils = require('jsx-ast-utils');
+
+const { elementType, getProp, getLiteralPropValue } = jsxAstUtils;
+
+const MESSAGE =
+  'The <va-telephone> Web Component should be used to display phone numbers.';
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    fixable: 'code',
+  },
+
+  create(context) {
+    return {
+      JSXElement(node) {
+        // Exit early if we aren't on an anchor link
+        if (elementType(node.openingElement) !== 'a') return;
+
+        const anchorNode = node.openingElement;
+
+        const hrefProp = getProp(anchorNode.attributes, 'href');
+        const hrefValue = getLiteralPropValue(hrefProp);
+
+        if (hrefValue?.startsWith('tel:')) {
+          const contact = hrefValue
+            .substring('tel:'.length)
+            .replace(/[^\d]/g, '');
```

Comment:
I decided against having this rule do any kind of length checking here. The [ESLint docs on applying fixes](https://eslint.org/docs/latest/developer-guide/working-with-rules#applying-fixes) say under best practices:

> Make fixes as small as possible. Fixes that are unnecessarily large could conflict with other fixes, and prevent them from being applied.

I didn't want to have a single rule rewrite code from an `<a>` to a `<va-telephone>` _and_ potentially truncate a phone number by removing some digits. I also didn't want to have this rule ignore `tel` links, so I wrote the second rule to take care of notifying developers when the `contact` prop doesn't have the right number of digits.